bondpy/bondpy/bondpy.py

```diff
@@ -313,20 +316,20 @@ def wait_until_formed(self, timeout=None):
             self.deadline = Timeout(timeout).reset()
         with self.lock:
             while self.sm.getState().getName() == 'SM.WaitingForSister':
-                if rospy.is_shutdown():
+                if not rclpy.ok(context=self.node.context):
                     break
-                if self.deadline and self.deadline.left() == rospy.Duration(0):
+                if self.deadline and self.deadline.left() == Duration(seconds=0):
                     break
                 wait_duration = 0.1
                 if self.deadline:
-                    wait_duration = min(wait_duration, self.deadline.left().to_sec())
+                    wait_duration = min(wait_duration, duration_to_sec(self.deadline.left()))
                 self.condition.wait(wait_duration)
             return self.sm.getState().getName() != 'SM.WaitingForSister'
 
     ## \brief Blocks until the bond is broken for at most 'duration'.
     #
     # \param timeout Maximum duration to wait.  If None then this call will not timeout.
-    # \return true iff the bond has been broken, even if it has never been formed.
+    # \return true if the bond has been broken, even if it has never been formed.
```

Comment:
```suggestion
    # \return true iff the bond has been broken, even if it has never been formed.
```

Reply:
Changed in [74805b1](https://github.com/ros/bond_core/pull/89/commits/74805b1f61bde433b7f39eaf3db5ecf52190c80f)